### PR TITLE
Adding educational card with link to videos UI modal

### PR DESCRIPTION
**Pull request by blackjackkent**
#### Changes proposed in this Pull Request

This PR adds a new card to all current "My Home" layouts with a link to a modal displaying the new videos UI.

This PR will be the central point of reference for our CFT for this feature, and should not be merged until the feature is fully approved. (It will probably require further updates as more modifications are made to the UI component being loaded in the modal.)

#### Testing instructions

* Check out D69391-code to your sandbox and sandbox the API. (Not necessary if D69391 has already been merged.)
* Check out this PR locally.
* Verify that the Calypso homepage now displays a card in the "education" carousel as follows:

image

* Verify that clicking on the link in the card opens a modal popup displaying the videos UI:

image

Related to #57326

### CODE REVIEW

File: client/my-sites/customer-home/cards/education/blogging-basics/index.jsx
Comment (kwight) [inline client/my-sites/customer-home/cards/education/blogging-basics/index.jsx]: It should be an em dash between "course" and "you'll".

**Comment by kwight** [comment]
This works well! I noted an em dash, and the graphic needs to be updated according to https://github.com/Automattic/wp-calypso/issues/57326#issuecomment-953497382.

The white modal chrome takes away from the dark background. 😕 Does this bother anyone the same? Should we go for dark chrome also, leave it, or maybe something else? cc/ @ollierozdarz @ianstewart

File: client/my-sites/customer-home/cards/education/blogging-basics/index.jsx
Comment (kwight) [reply client/my-sites/customer-home/cards/education/blogging-basics/index.jsx]: Oh, and spaces around the em dash: https://github.com/Automattic/wp-calypso/issues/57327#issuecomment-956506488

**Comment by jordesign** [comment]
> The white modal chrome takes away from the dark background. 😕 Does this bother anyone the same?

It kinda bothers me the same... Something a bit neater/more consistent would look like better attention to detail.
I like the idea of all black chrome - but happy to defer to other more informed opinions :)

**Comment by ollierozdarz** [comment]
Thanks for the ping. Are we able to please make it a full page modal as intended by the designs? I.e. No margin or padding on the outside and inside of the current modal (no bleed). I think this will do it its deserved justice!

We can replace the 'skip' link with the 'X' also. cc @kwight

**Comment by kwight** [comment]
> Are we able to please make it a full page modal as intended by the designs?

@ollierozdarz Ah, I might have been confused: I thought it was supposed to be a full screen thing in onboarding, but a modal in our first version for My Home. I'm not sure we've had a fullscreen modal before (usually they're just pages) – is the idea that it would be a modal in that the URL isn't changing? We'll see what we can do.

File: client/my-sites/customer-home/cards/constants.js
Comment (kwight) [inline client/my-sites/customer-home/cards/constants.js]: Let's update these to `home-education-blogging-quick-start`, to align with the actual course being called (see: p1635870290408700/1635868542.407800-slack-C029SB8JT8S).

**Comment by kwight** [comment]
@blackjackkent To do:

- [x] Correct card graphic.
- [x] Update constants.
- [x] Make the modal fullscreen.

**Comment by blackjackkent** [comment]
I've added a PR at https://github.com/Automattic/wp-calypso/pull/57572 that allows us to conditionally remove the links at the top of the UI (in this case, in favor of the Modal dismiss controls).

I'm going to pivot onto mobile styles for the component, but will come back to look at improving the cleanliness of the modal and reducing margin/padding as discussed.

File: client/my-sites/customer-home/cards/constants.js
Comment (blackjackkent) [reply client/my-sites/customer-home/cards/constants.js]: > Let's update these to `home-education-blogging-quick-start`, to align with the actual course being called (see: p1635870290408700/1635868542.407800-slack-C029SB8JT8S).

Added a diff at D69494-code to support this; will update this PR to match once it's merged.

**Comment by kwight** [comment]
@ianstewart A Tracks event for our Videos UI card is already handled by the existing events for the cards themselves. 👍 (And every view of the Videos UI, regardless of how it's triggered, is in https://github.com/Automattic/wp-calypso/pull/57584.)

**Comment by ollierozdarz** [comment]
> @ollierozdarz Ah, I might have been confused: I thought it was supposed to be a full screen thing in onboarding, but a modal in our first version for My Home. I'm not sure we've had a fullscreen modal before (usually they're just pages).

It's a valid question since this wasn't really in scope until quite recently (launching it from My Home). My reasoning for thinking it's better full screen with no bleed is because having it in a traditional modal just feels a little funny, and doesn't do the UI justice - I.e. the reason we have a dark UI is to draw people in, but if it *looks* like it's sitting in a modal/frame, it could provide more distraction.

Replicating the onboarding experience here (minus the skip link) will probably be the most ideal launching point, then we can iterate from there if needed.

File: client/my-sites/customer-home/cards/education/educational-content/index.jsx
Comment (kwight) [inline client/my-sites/customer-home/cards/education/educational-content/index.jsx:75]: Why was this necessary?..

**Comment by kwight** [comment]
@blackjackkent Can you double check why we seem to not have the button styles from https://github.com/Automattic/wp-calypso/pull/57580, and we also seem to have lost the course title (which shouldn't be Blogger Masterclass, but whatever comes from the API)?

File: client/my-sites/customer-home/cards/education/educational-content/index.jsx
Comment (blackjackkent) [reply client/my-sites/customer-home/cards/education/educational-content/index.jsx:75]: Because ModalComponent.displayName was undefined. (I had thought it was a general property of all React component objects but apparently not.) So we were getting a bunch of warnings in the console that the loop output did not have a key.

(I'm open to suggestions on a better value to use here.)

Comment (kwight) [reply client/my-sites/customer-home/cards/education/educational-content/index.jsx:75]: Ah, OK – that's fine. 👍

**Comment by blackjackkent** [comment]
> @blackjackkent Can you double check why we seem to not have the button styles from #57580, and we also seem to have lost the course title (which shouldn't be Blogger Masterclass, but whatever comes from the API)?

This seem to have been fixed by rebasing on the latest merged changes.

**Comment by blackjackkent** [comment]
Pushed updates to fix mdash and card graphic.

File: client/my-sites/customer-home/cards/education/blogging-basics/index.jsx
Comment (blackjackkent) [reply client/my-sites/customer-home/cards/education/blogging-basics/index.jsx]: Fixed.

**Comment by kwight** [comment]
@michaeldcain and @ianstewart pointed out that there's a `BlankCanvas` component that might help with the fullscreen "modal" effect we need: https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/components/blank-canvas/index.js

**Comment by jordesign** [comment]
Loving the way this is coming together! 😍  The fullscreen modal makes such a difference to the experience.

I know we're still tweaking but on my 16' Macbook - the spacious padding in the top elements mean that the full video player isn't loading before the bottom of the screen? A little thing - but I wonder if it is worth considering some visual tweaks to the padding?
Screen Shot 2021-11-10 at 8 15 11 am

**Comment by kwight** [comment]
Current behaviour, after the fullscreen and video load/play improvements.

**Comment by kwight** [comment]
> the spacious padding in the top elements mean that the full video player isn't loading before the bottom of the screen

Something to keep in mind, for sure. We're going to launch with the design we've got, and there will be another round of tweaks in the following sprint, I expect. cc/ @ollierozdarz @ianstewart

**Comment by andres-blanco** [comment]
I am getting `opacity: 1%` in the Play button when testing in the calypso.live site. 
image

On local testing this does not happen, I am disabling the opacity on the button for now.

**Review by andres-blanco — APPROVED** [review]

**Comment by andres-blanco** [comment]
Just in case, here's an undo PR: https://github.com/Automattic/wp-calypso/pull/58016

**Pull request merged**
